src/contract/warp-contract.ts

```diff
@@ -154,16 +154,22 @@ export class WarpContract implements ObserverContract {
   async writeInteraction(
     interaction: ObservationInteraction,
     options?: WriteInteractionOptions,
-  ): Promise<WriteInteractionResponse | null> {
+  ): Promise<WriteInteractionResponse> {
     await this.ensureContractInit();
 
     this.log.debug('Writing contract interaction...', { interaction });
     for (let i = 0; i < MAX_INTERACTION_RETRIES; i++) {
       try {
-        return await this.contract.writeInteraction(interaction, {
+        const response = await this.contract.writeInteraction(interaction, {
           disableBundling: true,
           ...options,
         });
+
+        if (!response) {
+          throw new Error();
```

Comment:
nit: we're not providing any message here. You could provide 'Failed to write interaction' or something to describe that warp did not directly throw but we did not get an interaction result

Reply:
I'll add `Failed to write interaction`.